rdt/transformers/text.py

```diff
@@ -0,0 +1,118 @@
+            do not replace them. Defaults to ``None``.
+        model_missing_values (bool):
+            Whether to create a new column to indicate which values were null or not. The column
+            will be created only if there are null values. If ``True``, create the new column if
+            there are null values. If ``False``, do not create the new column even if there
+            are null values. Defaults to ``False``.
+    """
+
+    DETERMINISTIC_TRANSFORM = False
+    DETERMINISTIC_REVERSE = False
+    INPUT_SDTYPE = 'text'
+    null_transformer = None
+
+    def __init__(self, regex_format='[A-Za-z]{5}', missing_value_replacement=None,
+                 model_missing_values=False):
+        self.data_length = None
+        self.missing_value_replacement = missing_value_replacement
+        self.model_missing_values = model_missing_values
+        self.regex_format = regex_format
+
+    def get_output_sdtypes(self):
+        """Return the output sdtypes supported by the transformer.
+
+        Returns:
+            dict:
+                Mapping from the transformed column names to supported sdtypes.
+        """
+        output_sdtypes = {}
+        if self.null_transformer and self.null_transformer.models_missing_values():
+            output_sdtypes['is_null'] = 'float'
+
+        return self._add_prefix(output_sdtypes)
+
+    def _fit(self, data):
+        """Fit the transformer to the data.
+
+        Args:
+            data (pandas.Series):
+                Data to fit to.
+        """
+        self.null_transformer = NullTransformer(
+            self.missing_value_replacement,
+            self.model_missing_values
+        )
+        self.null_transformer.fit(data)
+        self.data_length = len(data)
+
+    def _transform(self, data):
+        """Drop the column and return ``null`` column if ``models_missing_values``.
+
+        Args:
+            data (pandas.Series):
+                Data to transform.
+
+        Returns:
+            (numpy.ndarray or None):
+                If ``self.model_missing_values`` is ``True`` then will return a ``numpy.ndarray``
+                indicating which values should be ``nan``, else will return ``None``. In both
+                scenarios the original column is being dropped.
+        """
+        if self.null_transformer and self.null_transformer.models_missing_values():
+            return self.null_transformer.transform(data)[:, 1].astype(float)
+
+        return None
+
+    def _reverse_transform(self, data):
+        """Generate new data using the provided ``regex_format``.
+
+        Args:
+            data (pd.Series or numpy.ndarray):
+                Data to transform.
+
+        Returns:
+            pandas.Series
+        """
+        generator, size = strings_from_regex(self.regex_format)
+        if size > self.data_length:
+            reverse_transformed = np.array([
+                next(generator)
+                for _ in range(self.data_length)
```

Comment:
It seems like we are matching the number of rows specified by `self.data_length` which is set in `fit`. What happens if `reverse_transform` is called on a dataset with more or less rows than what we fitted? In the synthetic data case, the number of sampled rows is not necessarily the same as the number of rows used for modeling